File: catalog/backends/node-express/patterns/memory_leak_service.js
// Pattern: Profiling — gradual memory allocation in a cache
// Adapt: use a domain-relevant cache concept
const _cache = {};

function addToCache(req, res) {
  const key = req.query.key || 'default';
  _cache[key] = Buffer.alloc(1024 * 100); // 100KB per entry
  res.json({ cached: Object.keys(_cache).length, key });
}

module.exports = { addToCache };

File: catalog/backends/java-spring/patterns/MemoryLeakService.java
package com.example.service;

import org.springframework.web.bind.annotation.*;
import java.util.*;
import java.util.concurrent.ConcurrentHashMap;

// Pattern: Profiling — gradual memory allocation in a cache
// Adapt: use a domain-relevant cache concept
@RestController
public class MemoryLeakService {

    private final Map<String, byte[]> cache = new ConcurrentHashMap<>();

    @PostMapping("/api/cache")
    public Map<String, Object> addToCache(@RequestParam String key) {
        cache.put(key, new byte[1024 * 100]); // 100KB per entry
        return Map.of("cached", cache.size(), "key", key);
    }
}
